astrapy/exceptions.py

```diff
@@ -211,6 +211,77 @@ class DataAPIException(ValueError):
     pass
 
 
+@dataclass
+class DataAPIHttpException(DataAPIException, httpx.HTTPStatusError):
+    """
+    A request to the Data API resulted in an HTTP 4xx or 5xx response.
+
+    In most cases this comes with additional information: the purpose
+    of this class is to present such information in a structured way,
+    akin to what happens for the DataAPIResponseException, while
+    still raising (a subclass of) `httpx.HTTPStatusError`.
+
+    Attributes:
+        text: a text message about the exception.
+        error_descriptors: a list of all DataAPIErrorDescriptor objects
+            found in the response.
+    """
+
+    text: Optional[str]
+    error_descriptors: Optional[List[DataAPIErrorDescriptor]]
+
+    def __init__(
+        self,
+        text: Optional[str],
+        *,
+        httpx_error: httpx.HTTPStatusError,
+        error_descriptors: List[DataAPIErrorDescriptor],
```

Comment:
Should this be Optional here?

Reply:
Wouldn't `[]` convey the same information as `Nothing`? (***in this context***)

Reply:
Agree; and agree.
The Optional was an oversight from me, sorry and thank you for spotting this!
In line with the counterpart exception (DataAPIResponseException), there's no Optional and it is `[]` if nothing.

thanks!